src/main/java/com/thealgorithms/sorts/BitonicSort.java

```diff
@@ -1,79 +1,94 @@
 package com.thealgorithms.sorts;
 
-/* Java program for Bitonic Sort. Note that this program
-works only when size of input is a power of 2. */
-public class BitonicSort {
-
-    /* The parameter dir indicates the sorting direction,
-  ASCENDING or DESCENDING; if (a[i] > a[j]) agrees
-  with the direction, then a[i] and a[j] are
-  interchanged. */
-    void compAndSwap(int[] a, int i, int j, int dir) {
-        if ((a[i] > a[j] && dir == 1) || (a[i] < a[j] && dir == 0)) {
-            // Swapping elements
-            int temp = a[i];
-            a[i] = a[j];
-            a[j] = temp;
+import java.util.Arrays;
+
+/**
+ * BitonicSort class implements the SortAlgorithm interface using the bitonic sort technique.
+ */
+public class BitonicSort implements SortAlgorithm {
+
+    /**
+     * Sorts the given array using the Bitonic Sort algorithm.
+     *
+     * @param <T> the type of elements in the array, which must implement the Comparable interface
+     * @param arr the array to be sorted
+     * @return the sorted array
+     */
+    @Override
+    public <T extends Comparable<T>> T[] sort(T[] arr) {
+        if (arr == null || arr.length == 0) {
+            return arr;
         }
+
+        final int paddedSize = nextPowerOfTwo(arr.length);
+        T[] paddedArray = Arrays.copyOf(arr, paddedSize);
+
+        // Fill the padded part with a maximum value
+        final T maxValue = SortUtils.max(arr);
+        Arrays.fill(paddedArray, arr.length, paddedSize, maxValue);
+
+        bitonicSort(paddedArray, 0, paddedSize, true);
+        return Arrays.copyOf(paddedArray, arr.length);
     }
 
-    /* It recursively sorts a bitonic sequence in ascending
-  order, if dir = 1, and in descending order otherwise
-  (means dir=0). The sequence to be sorted starts at
-  index position low, the parameter cnt is the number
-  of elements to be sorted.*/
-    void bitonicMerge(int[] a, int low, int cnt, int dir) {
+    private <T extends Comparable<T>> void bitonicSort(T[] arr, int low, int cnt, boolean dir) {
```

Comment:
What do you think about introducing:
```java
private enum Direction {
    DESCENDING,
    ASCENDING,
}
```

Reply:
Agree. Code using enums is easier to read.